AURORA: Enable extraction of bitmaps from pe files

Nostritius opened the pull request:
This PR enables to extract bitmaps from pe exe files. It is thought as base for potentially implementing the spalash screens of dragonage and dragonage2.

Nostritius commented:
I have now rewritten the code to use another method for determining the type.

Nostritius commented:
The issues should now be fixed

DrMcCoy commented on src/aurora/pefile.cpp:
```diff
@@ -48,57 +48,86 @@ const Archive::ResourceList &PEFile::getResources() const {
+	for (iter = _resources.begin(); iter != _resources.end(); ++iter) {
+		if (iter->index == index)
+			break;
+	}
```
- If index isn't there in _resources, this will leave iter dangling and invoke UB
- Also, index is == the PE resource ID, no? Then you again have the issue that multiple IDs, but different types will clash

DrMcCoy commented on src/aurora/pefile.cpp:
```diff
@@ -121,6 +150,23 @@ void PEFile::load(const std::vector<Common::UString> &remap) {
+	for (std::vector<Common::PEResourceID>::const_iterator it = bitmapList.begin(); it != bitmapList.end(); ++it) {
+		Resource res;
+
+		uint32 id = it->getID() - 1;
```
If ID is 0, this will wrap. Why subtract 1 in the first place?

Nostritius commented:
I replaced now the pe id with a unique id for the archive and added a map which maps the unique id to the pe id. This should hopefully fix the problem of potential multiple ids with different types. I also fixed the other issues.

DrMcCoy commented:
The IDs you give them are sequential, no? Why does it need to be a map, then; shouldn't a vector be enough?

Nostritius commented:
You're right, I will change this

Nostritius commented:
This should now be done

DrMcCoy commented on src/aurora/pefile.cpp:
```diff
@@ -48,57 +48,86 @@ const Archive::ResourceList &PEFile::getResources() const {
-	// cursor format.
+	ResourceList::const_iterator iter = _resources.begin();
+	if (index >= _resources.size())
+		return 0;
```
This should probably even throw.

Compare with what the other archives do in this case: https://github.com/xoreos/xoreos/blob/master/src/aurora/rimfile.cpp#L97 , for example

Nostritius commented:
I have now replaced the return 0 through a throw.

DrMcCoy commented:
This looks good now, merged, thanks! :)
